src/java.base/share/classes/java/util/HashMap.java

```diff
@@ -2024,7 +2023,7 @@ final TreeNode<K,V> find(int h, Object k, Class<?> kc) {
                     p = pl;
                 else if (ph < h)
                     p = pr;
-                else if ((pk = p.key) == k || (k != null && k.equals(pk)))
+                else if (Objects.equals(k, (pk = p.key)))
```

Comment:
Stylistic: don't need parentheses around `pk = p.key`

Reply:
Stylistically, it calls attention to the assignment in the middle of an expression.